extendr-api/src/wrapper/list.rs

```diff
@@ -378,20 +378,17 @@ impl<T: AsRef<str>> KeyValue for (T, Robj) {
 }
 
 impl<T: Into<Robj>> FromIterator<T> for List {
-    /// Convert an iterator to a List object.
+    /// Convert an iterator to a `List` object.
     fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
-        crate::single_threaded(|| unsafe {
-            let values: Vec<SEXP> = iter
-                .into_iter()
-                .map(|s| Rf_protect(s.into().get()))
-                .collect();
+        let iter_collect: Vec<_> = iter.into_iter().collect();
+        let len = iter_collect.len();
 
-            let len = values.len();
+        crate::single_threaded(|| unsafe {
             let robj = Robj::alloc_vector(VECSXP, len);
-            for (i, v) in values.into_iter().enumerate() {
-                SET_VECTOR_ELT(robj.get(), i as isize, v);
+            for (i, v) in iter_collect.into_iter().enumerate() {
+                let item: Robj = v.into();
```

Comment:
```suggestion
                // We don't PROTECT each element here, as they will be immediately
                // placed into a list which will protect them:
                // https://cran.r-project.org/doc/manuals/R-exts.html#Garbage-Collection
                let item: Robj = v.into();
```

Reply:
This line yields a protected Robj by the ownership module.
This part is not controversial as everything is protected.
What isn't good is that it is clogging the ownership module.

Reply:
I was going to make a follow-up PR, where we create an unprotected Robj, (that also need to use the rust-conversion thing we have setup), and then deal with that.. I forgot what the plan would be there...